### ISSUE TITLE
About FDS in the test time

### ISSUE THREAD

**Issue by zhiqiiiiiii** [YyzHarry/imbalanced-regression#13]
Hi, thanks for the awesome job! I have a question about how FDS behaves during test time. It seems like that feature smoothing is disabled during the test since FDS module is only called during training for agedb-dir as follows:
```     
if self.training and self.fds:
            if epoch >= self.start_smooth:
                encoding_s = self.FDS.smooth(encoding_s, targets, epoch)
```
Could you explain why you don't do feature smooth during testing?

**Comment by YyzHarry** [comment]
Thanks for your interest. FDS requires labels to do smoothing over nearby targets (see the `targets` argument), which is not available during inference.

Nevertheless, adding FDS module during training can implicitly regularize the model to learn to generate features that are more accurate even without smoothing, and finally the smoothing module can be removed during inference (the L1 distance between the running statistics and the smoothed statistics during training will gradually diminish to near zero). More details can be found in the [paper](https://arxiv.org/pdf/2102.09554.pdf) on page 8, "Understanding FDS" part (and Figure 8).
